### PR TITLE
Send respawn packet by default

### PR DESCRIPTION
**Pull request by mathiascode**
I don't see why the respawn packet is disabled by default. Not sending one breaks the client, and causes confusion for plugin developers.

### CODE REVIEW

**Comment by peterbell10** [comment]
Lua API docs also need to be updated (or should we just deprecated the argument from Lua?)

File: src/NetherPortalScanner.cpp [hunk 1]
```diff
@@ -296,7 +296,7 @@ void cNetherPortalScanner::OnDisabled(void)
 	}
 
 	FLOGD("Placing player at {0}", Position);
-	m_Entity->MoveToWorld(m_World, Position, true);
+	m_Entity->MoveToWorld(m_World, Position, true, false);
```
Comment (peterbell10) [inline src/NetherPortalScanner.cpp]: There are also the end portal cases to consider.

Comment (mathiascode) [reply src/NetherPortalScanner.cpp]: As far as I can tell, these cases already disable respawn packets explicitly.

**Comment by mathiascode** [comment]
Easier to just leave it in in my opinion.

File: src/NetherPortalScanner.cpp (hunk 1 above)
Comment (peterbell10) [reply src/NetherPortalScanner.cpp]: Ah you're right, I was forgetting there are multiple signatures.

Comment (mathiascode) [reply src/NetherPortalScanner.cpp]: I wish there weren't so many, but what's done is done.

**Comment by madmaxoft** [comment]
I believe the respawn packet should only be sent when changing dimensions. If you travel from one overworld to another, you must not send a respawn packet.

**Comment by mathiascode** [comment]
Afaik, it's always sent in vanilla (or at least some equivalent to it). If you don't send it, the client jitters around for a bit until the player has teleported, even if the dimension is the same.

**Pull request merged**